src/store.js

```diff
@@ -1,12 +1,16 @@
 import MiniPromise from './ponyfills/minipromise.js';
 import * as objects from './ponyfills/objects.js';
 import * as strings from './ponyfills/strings.js';
+import * as arrays from './ponyfills/arrays.js';
 
 import { evaluate } from './predicates.js';
 import { waitFor, emit } from './waitforit.js';
 import { CONTEXT_CHANGED } from './context.js';
 import retrieve from './retrieve.js';
 
+// Lock for updates, is updated by reevaluateContext()
+let locked = false;
```

Comment:
You might want to be more verbose in the the naming of this. Something like: alreadyReevaluatingContext.